src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -148,18 +164,62 @@ class FilterButton extends React.Component<Props, State> {
         }
     };
 
-    deleteCondition = (index: number) => {
-        const { conditions } = this.state;
+    updateSelectedPrefix = (option: SelectOptionType) => {
+        const prefix = option.displayText;
 
-        const updatedConditions = conditions.filter((condition, conditionIndex) => {
-            return conditionIndex !== index;
+        this.setState({
+            selectedPrefix: prefix,
+        });
+
+        this.updateConditionsPrefixes(prefix);
+    };
+
+    updateConditionsPrefixes = (prefix: string) => {
+        const { conditions } = this.state;
+        const conditionsAfterDeletion = conditions.map((condition, index) => {
+            if (index !== 0) {
+                const updatedCondition = Object.assign({}, condition);
+                updatedCondition.prefix = prefix;
+                return updatedCondition;
+            }
+            return condition;
         });
 
+        // The first condition must always have a prefix of WHERE.
+        const firstCondition = Object.assign({}, conditionsAfterDeletion[0]);
+        firstCondition.prefix = WHERE;
+
+        const updatedConditions = [firstCondition, ...conditionsAfterDeletion.slice(1)];
+
         this.setState({
             conditions: updatedConditions,
         });
     };
 
+    deleteCondition = (index: number) => {
+        const { conditions } = this.state;
+
+        const conditionsAfterDeletion = conditions.filter((condition, conditionIndex) => {
+            return conditionIndex !== index;
+        });
+
+        if (conditionsAfterDeletion.length === 0) {
+            this.setState({
+                conditions: [],
+            });
+        } else {
+            // The first condition must always have a prefix of WHERE.
```

Comment:
I'm not too familiar with how these conditions work but this arrangement feels brittle. Let's talk in person about how this is structured.